status-im/status-react
# [#5749] transaction history fixes

partially addresses #5749 

### Summary:

Restricted the range of blocks where we get the event logs for token transfers to max 100000 in order to get around the infura timeouts.

What this means is that we're only scanning for about last two weeks of transactions, but at least they will appear in the history.

This PR also fixes several bugs:
* the transaction polling process could easily die and not restart
* the confirmations were not being merged correctly leading to strange cases where a transaction is displayed as not confirmed when in fact it was
* prevented latent transaction queries from a different account from interfering with a newly changed account
* stopped continuous polling when transactions are present from other network in chat
* fixed miss-formatted url

## Possible follow up from here

To get the older token transfers we can recursively call the same query with a moving 100k blocks large window all the way down to the block where the oldest token was created (possibly dropping tokens as we go beyond their own creation block). This would be a network intensive operation, so I don't want to do that every time user refreshes the list. Instead this should be done on account recovery. Will be provided in a separate PR.

### Testing notes (optional):

Only token transfers from the last 100k blocks (about two weeks) will appear.

#### Areas that maybe impacted (optional)
**Functional**
- wallet / transactions

### Steps to test:
- Open Wallet on Mainnet or Ropsten
- Send some tokens to this phone
- Check the transaction history

status: ready
